code/ship/shiphit.cpp

```diff
@@ -72,6 +73,12 @@ vec3d	Original_vec_to_deader;
 
 static bool global_damage = false;
 
+const std::shared_ptr<scripting::OverridableHook> OnPainFlashHook = scripting::OverridableHook::Factory(
+	"On Pain Flash", "Called when a pain flash is displayed.",
+	{ 		
+		{"Pain_Type", "number", "The type of pain flash displayed: shield = 0 and hull = 1."},
+	}, CHA_PAINFLASH);
```

Comment:
Maybe the on frame hook wasn't ideal here. You don't need to add a new `CHA_` entry. The new system takes care of that for you.